pandas Dataframe with 1 row results in AssertionError during call to validate

**Describe the bug**

When calling *validate* for a SchemaModel on a pandas's data frame with only 1 row, an AssertionError is raised. This appears to be triggered when coerce=True is includex in the field definition, but seems like a simple fix as detailed below

- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the latest version of pandera.
- [x] (optional) I have confirmed this bug exists on the master branch of pandera.

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
from pandera.typing import Series
import pandas as pd
import pandera as pa

class Model(pa.SchemaModel):
    x: Series[int] = pa.Field(coerce=True)

Model.validate(pd.DataFrame({"x": [None]}))

# Results in the following error
# ...
# File "/cluster/home/willems/MethodsDev/vcgt/garbnewenv/pandera-fix/lib/python3.9/site-packages/pandera/engines/numpy_engine.py", line 57, in coerce
#    failure_cases=utils.numpy_pandas_coerce_failure_cases(
#  File "/cluster/home/willems/MethodsDev/vcgt/garbnewenv/pandera-fix/lib/python3.9/site-# packages/pandera/engines/utils.py", line 88, in numpy_pandas_coerce_failure_cases
#    check_output = numpy_pandas_coercible(data_container, type_)
#  File "/cluster/home/willems/MethodsDev/vcgt/garbnewenv/pandera-fix/lib/python3.9/site-packages/pandera/engines/utils.py", line 33, in numpy_pandas_coercible
#    search_list = _bisect(series)
#   File "/cluster/home/willems/MethodsDev/vcgt/garbnewenv/pandera-fix/lib/python3.9/site- packages/pandera/engines/utils.py", line 20, in _bisect
#    assert (
# AssertionError: cannot bisect a pandas Series of length < 2

```

#### Expected behavior
The above validation should fail with a SchemaError instead of an AssertionError

#### Desktop (please complete the following information):
python 3.9.1
pandas             1.3.1
pandera            0.7.1


#### Additional context

Seems like the [numpy_pandas_coercible() fn](https://github.com/pandera-dev/pandera/blob/master/pandera/engines/utils.py#L13) is missing a base case for a series that contains only 1 element on line 33:

```python
# Current call which triggers the AssertionError as _bisect immediate checks that series.size >= 2
search_list = _bisect(series)

# Proposed fix
search_list =[[series.iloc[0]] if series.size == 1 else _bisect(series)

# I'm not sure if adding a check before line 33 for an empty series is also necessary in case this function is (mis)called on an empty series
if series.empty:
   return pd.Series(dtype="bool")

```

--- a/pandera/engines/utils.py
+++ b/pandera/engines/utils.py
@@ -16,9 +16,6 @@ def numpy_pandas_coercible(series: pd.Series, type_: Any) -> pd.Series:
     Bisects the series until all the failure cases are found.
     """
 
-    if series.empty:
-        return pd.Series(dtype=bool)
-
     def _bisect(series):
         assert (
             series.shape[0] >= 2

--- a/pandera/engines/utils.py
+++ b/pandera/engines/utils.py
@@ -16,6 +16,9 @@ def numpy_pandas_coercible(series: pd.Series, type_: Any) -> pd.Series:
     Bisects the series until all the failure cases are found.
     """
 
+    if series.empty:
+        return pd.Series(dtype=bool)
+
     def _bisect(series):
         assert (
             series.shape[0] >= 2
@@ -30,7 +33,7 @@ def numpy_pandas_coercible(series: pd.Series, type_: Any) -> pd.Series:
         except (ValueError, TypeError):
             return False
 
-    search_list = _bisect(series)
+    search_list = [series] if series.size == 1 else _bisect(series)
     failure_index = []
     while search_list:
         candidates = []

--- a/tests/core/test_schemas.py
+++ b/tests/core/test_schemas.py
@@ -101,6 +101,12 @@ def test_dataframe_single_element_coerce() -> None:
         schema(pd.DataFrame({"x": [None]}))
 
 
+def test_dataframe_empty_coerce() -> None:
+    """Test that coercing an empty element dataframe works correctly."""
+    schema = DataFrameSchema({"x": Column(int, coerce=True)})
+    assert isinstance(schema(pd.DataFrame({"x": []})), pd.DataFrame)
+
+
 def test_dataframe_schema_equality() -> None:
     """Test DataframeSchema equality."""
     schema = DataFrameSchema({"a": Column(int)})

--- a/tests/core/test_schemas.py
+++ b/tests/core/test_schemas.py
@@ -93,6 +93,14 @@ def test_dataframe_schema() -> None:
         schema.validate(df.assign(a=[1.7, 2.3, 3.1]))
 
 
+def test_dataframe_single_element_coerce() -> None:
+    """Test that coercing a single element dataframe works correctly."""
+    schema = DataFrameSchema({"x": Column(int, coerce=True)})
+    assert isinstance(schema(pd.DataFrame({"x": [1]})), pd.DataFrame)
+    with pytest.raises(errors.SchemaError, match="Error while coercing 'x'"):
+        schema(pd.DataFrame({"x": [None]}))
+
+
 def test_dataframe_schema_equality() -> None:
     """Test DataframeSchema equality."""
     schema = DataFrameSchema({"a": Column(int)})

